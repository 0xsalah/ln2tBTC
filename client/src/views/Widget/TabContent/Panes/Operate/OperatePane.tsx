import React, { useState, useEffect } from "react";
import ICONS from "../../../../../img/icons.svg";
import ActionButton from "../../../common/ActionButton";
import Input from "../../../common/Input";
import Notification from "../../../common/Notification";
import { toMaxDecimalsRound } from "../../../utils";

interface OperatePaneProps {
  handleInputChange?: (form: { [key: string]: string | number }) => void;
}

const OperatePane: React.FC<OperatePaneProps> = (props) => {
  const { handleInputChange = () => null } = props;

  const [formValues, setFormValues] = useState({
    lnBalance: "",
    tbtcBalance: "",
    linearFee: "",
    constantFee: "",
    nodeAddress: "",
  });

  const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
    //setXAmount is the amount displayed in the input, should be string
    const name = ev.target.name;
    let value = ev.target.value;
    if (ev.target.type === "number")
      value =
        ev.target.value === ""
          ? ev.target.value
          : toMaxDecimalsRound(ev.target.value, +ev.target.step).toString();

    setFormValues((old) => {
      return {
        ...old,
        [name]: value,
      };
    });
  };

  useEffect(() => {
    handleInputChange(formValues);
  }, [formValues, handleInputChange]);

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__operate operate">
            <div className="operate__form">
              <Notification>
                <span>Add liquidity and earn fees on user swaps. <a href="/">Learn more</a>.</span>
              </Notification>
              <div className="form-group">
                <Input
                  value={formValues["lnBalance"]}
                  onChange={onChange}
                  name="lnBalance"
                  label="LN Balance "
                  svgIcon={
                    <svg className="icon icon-flash">
                      <use xlinkHref={`${ICONS}#icon-flash`}></use>
                    </svg>
                  }
                  placeholder="0.0"
                  type="number"
                />
              </div>
              <div className="form-group">
                <Input
                  value={formValues["tbtcBalance"]}
                  onChange={onChange}
                  name="tbtcBalance"
                  label="tBTC Balance "
                  svgIcon={
                    <svg className="icon icon-man">
                      <use xlinkHref={`${ICONS}#icon-man`}></use>
                    </svg>
                  }
                  placeholder="0.0"
                  type="number"
                />
              </div>
              <div className="form-group">
                <Input
                  value={formValues["linearFee"]}
                  onChange={onChange}
                  name="linearFee"
                  label="Linear Fee "
                  placeholder="0.0"
                  type="number"
                  hint={
                    <>
                      <b>Linear Fee</b> is a format for a Lightning Network
                      invoice uses a bech32 encoding, which is also used for
                      Bitcoin’s Segregated Witness. <br />
                      <a href="/">Learn more</a>
                    </>
                  }
                />
              </div>
              <div className="form-group">
                <Input
                  value={formValues["constantFee"]}
                  onChange={onChange}
                  name="constantFee"
                  label="Constant Fee "
                  placeholder="0.0"
                  type="number"
                  hint={
                    <>
                      <b>Linear Fee</b> is a format for a Lightning Network
                      invoice uses a bech32 encoding, which is also used for
                      Bitcoin’s Segregated Witness. <br />
                      <a href="/">Learn more</a>
                    </>
                  }
                />
              </div>
              <div className="form-group">
                <Input
                  value={formValues["nodeAddress"]}
                  onChange={onChange}
                  name="nodeAddress"
                  label="Node Address "
                  placeholder="0.0"
                  type="number"
                  hint={
                    <>
                      <b>Linear Fee</b> is a format for a Lightning Network
                      invoice uses a bech32 encoding, which is also used for
                      Bitcoin’s Segregated Witness. <br />
                      <a href="/">Learn more</a>
                    </>
                  }
                />
              </div>
              <ActionButton text="Register" className="operate__button" />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default OperatePane;
