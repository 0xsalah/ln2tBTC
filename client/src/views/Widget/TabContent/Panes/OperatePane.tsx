import React from 'react'
import ICONS from '../../../../img/icons.svg'
import ActionButton from '../../common/ActionButton'
import Input from '../../common/Input'
import Notification from '../../common/Notification'

const OperatePane: React.FC = () => {


    return (
        <div className="tab-pane is_active">
            <div className="tab-pane__content">
                <div className="box-operation__content">
                    <div className="box-operation__operate operate">
                        <form action="#" className="operate__form">
                            <Notification>
                                Add liquidity and earn fees on user swaps. <a href="/">Learn more</a>
                            </Notification>
                            <div className="form-group">
                                <Input
                                    label="LN Balance "
                                    svgIcon={
                                        <svg className="icon icon-flash">
                                            <use xlinkHref={`${ICONS}#icon-flash`}></use>
                                        </svg>
                                    }
                                    placeholder="0.0"
                                    type="text"
                                />
                            </div>
                            <div className="form-group">
                                <Input
                                    label="tBTC Balance "
                                    svgIcon={
                                        <svg className="icon icon-man">
                                            <use xlinkHref={`${ICONS}#icon-man`}></use>
                                        </svg>
                                    }
                                    placeholder="0.0"
                                    type="text"
                                />
                            </div>
                            <div className="form-group">
                                <Input
                                    label="Linear Fee "
                                    placeholder="0.0"
                                    type="text"
                                    hint={(
                                        <>
                                            <b>Linear Fee</b> is a format for a Lightning Network invoice uses a bech32 encoding, which is also used for Bitcoin’s Segregated Witness. <br />
                                            <a href="/">Learn more</a>
                                        </>
                                    )}
                                />
                            </div>
                            <div className="form-group">
                                <Input
                                    label="Constant Fee "
                                    placeholder="0.0"
                                    type="text"
                                    hint={(
                                        <>
                                            <b>Linear Fee</b> is a format for a Lightning Network invoice uses a bech32 encoding, which is also used for Bitcoin’s Segregated Witness. <br />
                                            <a href="/">Learn more</a>
                                        </>
                                    )}
                                />
                            </div>
                            <div className="form-group">
                                <Input
                                    label="Node Address "
                                    placeholder="0.0"
                                    type="text"
                                    hint={(
                                        <>
                                            <b>Linear Fee</b> is a format for a Lightning Network invoice uses a bech32 encoding, which is also used for Bitcoin’s Segregated Witness. <br />
                                            <a href="/">Learn more</a>
                                        </>
                                    )}
                                />
                            </div>
                            <ActionButton text="Register" className="operate__button" />
                        </form>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default OperatePane