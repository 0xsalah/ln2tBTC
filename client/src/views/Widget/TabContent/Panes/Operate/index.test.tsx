import React from "react";
import { render } from "@testing-library/react";
import OperatePane from ".";

test("snapshot of component rendered", () => {
  const component = render(<OperatePane />);
  expect(component.baseElement).toMatchSnapshot();
});
