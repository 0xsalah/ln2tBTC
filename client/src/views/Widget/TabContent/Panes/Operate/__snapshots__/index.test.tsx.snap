// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered 1`] = `
<body>
  <div>
    <div
      class="tab-pane is_active"
    >
      <div
        class="tab-pane__content"
      >
        <div
          class="box-operation__content"
        >
          <div
            class="box-operation__operate operate"
          >
            <div
              class="operate__form"
            >
              <div
                class="notification "
              >
                <div
                  class="notification__text"
                >
                  Add liquidity and earn fees on user swaps. 
                  <a
                    href="/"
                  >
                    Learn more
                  </a>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="exchange__column "
                >
                  <div
                    class="form-block "
                  >
                    <label
                      class="form-block__label"
                    >
                      <span>
                        LN Balance 
                      </span>
                    </label>
                    <div
                      class="form-block__box"
                    >
                      <input
                        aria-label="input"
                        class="form-block__input form-control"
                        name="lnBalance"
                        placeholder="0.0"
                        type="number"
                        value=""
                      />
                      <div
                        class="form-block__icon"
                      >
                        <svg
                          class="icon icon-flash"
                        >
                          <use
                            xlink:href="icons.svg#icon-flash"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="exchange__column "
                >
                  <div
                    class="form-block "
                  >
                    <label
                      class="form-block__label"
                    >
                      <span>
                        tBTC Balance 
                      </span>
                    </label>
                    <div
                      class="form-block__box"
                    >
                      <input
                        aria-label="input"
                        class="form-block__input form-control"
                        name="tbtcBalance"
                        placeholder="0.0"
                        type="number"
                        value=""
                      />
                      <div
                        class="form-block__icon"
                      >
                        <svg
                          class="icon icon-man"
                        >
                          <use
                            xlink:href="icons.svg#icon-man"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="exchange__column "
                >
                  <div
                    class="form-block "
                  >
                    <label
                      class="form-block__label"
                    >
                      <span>
                        Linear Fee 
                      </span>
                      <span
                        class="tooltip-help"
                      >
                        <svg
                          class="icon icon-question"
                        >
                          <use
                            xlink:href="icons.svg#icon-question"
                          />
                        </svg>
                      </span>
                    </label>
                    <div
                      class="form-block__box"
                    >
                      <input
                        aria-label="input"
                        class="form-block__input form-control"
                        name="linearFee"
                        placeholder="0.0"
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="exchange__column "
                >
                  <div
                    class="form-block "
                  >
                    <label
                      class="form-block__label"
                    >
                      <span>
                        Constant Fee 
                      </span>
                      <span
                        class="tooltip-help"
                      >
                        <svg
                          class="icon icon-question"
                        >
                          <use
                            xlink:href="icons.svg#icon-question"
                          />
                        </svg>
                      </span>
                    </label>
                    <div
                      class="form-block__box"
                    >
                      <input
                        aria-label="input"
                        class="form-block__input form-control"
                        name="constantFee"
                        placeholder="0.0"
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="exchange__column "
                >
                  <div
                    class="form-block "
                  >
                    <label
                      class="form-block__label"
                    >
                      <span>
                        Node Address 
                      </span>
                      <span
                        class="tooltip-help"
                      >
                        <svg
                          class="icon icon-question"
                        >
                          <use
                            xlink:href="icons.svg#icon-question"
                          />
                        </svg>
                      </span>
                    </label>
                    <div
                      class="form-block__box"
                    >
                      <input
                        aria-label="input"
                        class="form-block__input form-control"
                        name="nodeAddress"
                        placeholder="0.0"
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="operate__button"
              >
                <button
                  class="btn btn--full btn--primary"
                >
                  <span>
                    Register
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
